protocols/kad/src/behaviour.rs

```diff
@@ -476,13 +499,19 @@ where
     ///
     /// > **Note**: Bootstrapping requires at least one node of the DHT to be known.
     /// > See [`Kademlia::add_address`].
-    pub fn bootstrap(&mut self) {
+    pub fn bootstrap(&mut self) -> Result<QueryId, ()> {
         let local_key = self.kbuckets.local_key().clone();
-        let info = QueryInfo::Bootstrap { peer: local_key.preimage().clone() };
+        let info = QueryInfo::Bootstrap {
+            peer: local_key.preimage().clone(),
+            remaining: Vec::new().into_iter()
```

Comment:
Instead of modelling the uninitialized state with an empty vector, I would find an `Option<Vec<_>>::None` easier to understand, what do you think?

Reply:
What is the meaning of `Some(Vec::new())` then? Wouldn't we add another type inhabitant that is never supposed to be instantiated? `Option<NonEmptyVec<_>>` would be a good isomorphic type but I don't think it is worth introducing such a thing here, though I would generally like it if we had non-empty collection types for many purposes.

Reply:
Oh, I think I misunderstood. You suggest intentionally distinguishing `Some(Vec::new())` from `None` because the query in this state may be visible to the user. I wrongly remembered this to only be an internal intermediate state. That makes sense, thanks for the suggestion!